toptal/picasso
# fix: rich text editor behaviour in forms

[FX-4105]

### Description

This pull request fixes bug with Rich Text Editor blocking the form, so the RTE field does not lose focus (bug was introduced in https://github.com/toptal/picasso/pull/3645).

The solution is to move the `registerNodeTransform()` in the separate plugin.

### How to test

Steps to reproduce

- check out https://github.com/toptal/picasso/commits/feature/rte-lexical
- run `yarn start`
- navigate to ?path=/story/picasso-forms-form--form
- focus into RTE field, then focus into any other field in other stories
- the focus will stay in RTE

Navigate to https://picasso.toptal.net/FX-NULL-fix-rte-in-forms/?path=/story/picasso-forms-form--form and test it the same way

### Screenshots

🔴 Before 

https://github.com/toptal/picasso/assets/1390758/aad3afee-cf63-4960-bea6-24920165f1e4

🟢 After 

https://github.com/toptal/picasso/assets/1390758/c6ca8b87-8a74-4349-a590-12e44b7daedf

### Development checks

- [N/A] Add changeset according to [guidelines](https://github.com/toptal/picasso/blob/master/docs/contribution/changeset-guidelines.md) (if needed)
- [N/A] Make sure that additions and changes on design follow [Toptal's BASE design](https://design.toptal.net/), and it's been already discussed with designers at #-base-core
- [N/A] Annotate all `props` in component with documentation
- [N/A] Create `examples` for component
- [N/A] Ensure the changed/created components have not caused accessibility issues. [How to use accessibility plugin in storybook](https://github.com/toptal/picasso/blob/master/docs/contribution/accessibility.md).

> All **_development checks_** should be done and set checked to pass the
> **GitHub Bot: TODOLess** action

<details>
<summary>PR commands</summary>
<br />

List of available commands:

- `@toptal-bot run package:alpha-release` - Release alpha version
- `@toptal-anvil ping reviewers` - Ping FX team for review

</details>

<details>
<summary>PR Review Guidelines</summary>
<br />

#### When to approve? ✅

**You are OK** with merging this PR and

1. You have no extra requests.
2. You have optional requests.
   1. Add `nit:` to your comment. (ex. `nit: I'd rename this variable from makeCircle to getCircle`)

#### When to request changes? ❌

**You are not OK** with merging this PR because

1. Something is broken after the changes.
2. Acceptance criteria is not reached.
3. Code is dirty.

#### When to comment (neither ✅ nor ❌)

**You want your comments to be addressed** before merging this PR in cases like:

1. There are leftovers like unnecessary logs, comments, etc.
2. You have an opinionated comment regarding the code that requires a discussion.
3. You have questions.

#### How to handle the comments?

1. An owner of a comment is the only one who can resolve it.
2. An owner of a comment must resolve it when it's addressed.
3. A PR owner must reply with ✅ when a comment is addressed.

</details>


[FX-4105]: https://toptal-core.atlassian.net/browse/FX-4105?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
